middlewares/auth.middleware.js

```diff
@@ -249,6 +272,66 @@ function deleteResetToken(req, res, next) {
     );
 }
 
+/**
+ * Attempts to find a rolebinding given the role name and adds it to the account
+ * @param {ObjectId} accountId the id of the account that you want to add a rolebinding to
+ * @param {String} roleName the name of the role that you want to add
+ */
+async function createRoleBindingByRoleName(accountId, roleName){
+    const role = await Services.Role.getRole(roleName);
+    if (!!role) {
+        await Services.RoleBinding.createRoleBinding(accountId, role.id);
+    }
+}
+
+/**
+ * Middleware that creates rolebinding to access POST route for respective account
+ * @param {{body: {account:{accountType:String, id: ObjectId}}}} req the request object
+ * @param {*} res 
+ * @param {(err?)=>void} next 
+ */
+async function addCreationRoleBindings(req, res, next){
+    // Get the default role for the account type given
+    const roleName = Constants.POST_ROLES[req.body.account.accountType];
+    await createRoleBindingByRoleName(req.body.account.id, roleName);
+    next();
+}
+
+/**
+ * Adds proper account rolebindings on account creation
+ * @param {string} roleName name of the role to be added to account
+ */
+function addRoleBindings(roleName){
+    return async (req, res, next) => {
+        if(!!req.body.hackerDetails){
```

Comment:
If we're looking for .accountId, that should be the same as the req.user.id right? Can we use that instead of searching through all the _____Details? Though using req.user.id may cause an issue if we need to create some type without having the user be logged in. Nevertheless, I'm a little hesitant about having to check through all the ____Details for creation from a maintenance and extensibility perspective.

Reply:
Didn't realize I could do this, thank you!